.clang-tidy

```diff
@@ -1,6 +1,6 @@
 ---
 # turn off readability-braces-around-statements to allow single line statement like 'if (x == y) doSomething();'
-Checks:          '-*,cppcoreguidelines-*,google-*,readability-*,modernize-*,-readability-braces-around-statements,-google-runtime-references,-cppcoreguidelines-pro-type-reinterpret-cast'
+Checks:          '-*,cppcoreguidelines-*,google-*,readability-*,modernize-*,bugprone-*,-readability-braces-around-statements,-google-runtime-references,-cppcoreguidelines-pro-type-reinterpret-cast'
```

Comment:
`.clang-tidy` file is a `yaml` file literally, if desired, you can actually make it mulit-line by using a yaml multiple line string, and `clang-tidy` itself happens to tolerate the `\n` tokens.

Reply:
Done. Thanks for the suggestion. Do you have any thoughts on the kind of checks we should enable? Discussion happened in https://github.com/microsoft/onnxruntime/pull/12655 but was inconclusive.

@snnn @edgchen1

Reply:
FYI, I've passed this question along to the core team.

Reply:
@justinchuby I don't have suggestions to enable more check, instead, I'd suggest to disable more.

1. `modernize-concat-nested-namespaces`, becaure we don't use it.
2. `cppcoreguidelines-macro-usage`, there is a lot of false-positive
    > Function-like macro 'Foo' used; consider a 'constexpr' template function
3. `readability-identifier-length`, clang-tidy is bullshitting here, there are a lot of **numerical code** rely on short names to imporve readability.
4. `google-readability-todo`, Missing username/bug in TODO which I am not sure we are practicing.
5. `cppcoreguidelines-pro-bounds-array-to-pointer-decay` Do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast instead. But our error checking macro `ORT_RETURN_IF` and `LOGS` is relying on it. We shall fix it, either way.

hint, `onnxruntime/core/session/inference_session.cc` is full of warning in current setting xDDD

Reply:
@cloudhan I added all except for `cppcoreguidelines-pro-bounds-array-to-pointer-decay`, because it does seem desired from my understanding of your explanation, even though some of our macros have this behavior?

Reply:
Removed `-google-readability-braces-around-statements` because `google-readability-braces-around-statements.ShortStatementLines` is set to 1.

Reply:
Yeah, that is OK to keep `cppcoreguidelines-pro-bounds-array-to-pointer-decay`, it has its purpose. But I think someone should fix it from source code level. And if we add the magic comment to disable it at the macro definition side, I am not sure if it has effect on macro application site.